projects/standard-rulesets/src/spectral/index.ts

```diff
@@ -0,0 +1,105 @@
+import { SpectralRule } from '@useoptic/rulesets-base';
+import { Ruleset, Spectral } from '@stoplight/spectral-core';
+import fs from 'fs';
+import fetch from 'node-fetch';
+// @ts-ignore
+import { bundleAndLoadRuleset } from '@stoplight/spectral-ruleset-bundler/with-loader';
+import Ajv from 'ajv';
+import { IChange, IFact, OpenAPIV3, Result } from '@useoptic/openapi-utilities';
+// import fetch from 'node-fetch';
+
+const ajv = new Ajv();
+const configSchema = {
+  type: 'object',
+  properties: {
+    added: {
+      type: 'string',
+    },
+    always: {
+      type: 'array',
+      items: {
+        type: 'string',
+        description: 'URI of spectral ruleset file (file or URL)',
+      },
+    },
+  },
+};
+const validateConfigSchema = ajv.compile(configSchema);
+
+export class SpectralRulesets {
+  constructor(
+    private options: {
+      spectralRules: Promise<SpectralRule>[];
+    }
+  ) {}
+
+  async preparedRulesets() {
+    return await Promise.all(this.options.spectralRules);
+  }
+
+  async runRules(inputs: {
+    context: any;
+    nextFacts: IFact[];
+    currentFacts: IFact[];
+    changelog: IChange[];
+    nextJsonLike: OpenAPIV3.Document<{}>;
+    currentJsonLike: OpenAPIV3.Document<{}>;
+  }): Promise<Result[]> {
+    const rulesets = await this.preparedRulesets();
```

Comment:
some choices to make here...

right now if even 1 of the rulesets you try to load fails you'll get an error as you run the rules. If we push that error to the fromOpticConfig it'll fail before trying to run, but those have currently been sync and this would make them async

We could swallow errors and warn if any of these failed to load or run. I think it's fine to fail them all right now, but was curious what @niclim imagined for rulessets that fail to stand up or run

Reply:
done